units/src/amount.rs

```diff
@@ -196,25 +221,50 @@ pub enum ParseAmountError {
     /// The amount is too big or too small.
     OutOfRange(OutOfRangeError),
     /// Amount has higher precision than supported by the type.
-    TooPrecise,
-    /// Invalid number format.
-    InvalidFormat,
+    TooPrecise(TooPreciseError),
+    /// A digit was expected but not found.
+    MissingDigits(MissingDigitsError),
     /// Input string was too large.
-    InputTooLarge,
+    InputTooLarge(InputTooLargeError),
     /// Invalid character in input.
-    InvalidCharacter(char),
+    InvalidCharacter(InvalidCharacterError),
```

Comment:
Should we shorten this to `InvalidCharError` just for our sanity because we have that in other places (`hex` at least)?

Reply:
Good question. `Character` is a tiny bit more readable than `Char` IMO but maybe shorter is better. Though the code is already written and most people will not refer to the variant anyway.

Reply:
The error types are already long so if you think `Character` is more readable that `Char` we can leave this one as is and when I next touch `InvalidCharError` I'll change it to use the long form.